### DISCUSSION TITLE
Support Security Scheme

### DISCUSSION THREAD

**Discussion by ahx** [ahx/openapi_first#171] [General]
We currently just ignore [security schemes](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#securitySchemeObject). Let's change that.

**Comment by muhammadn** [comment]
@ahx I'm interested to know more about the progress about this feature.

**Comment by ahx** [comment]
Hi @muhammadn, I am not working on handling security schemes right now. Can you outline how you would expect this to be handled in a rack middleware?
I know that [connexion](https://github.com/spec-first/connexion) (python) handles this somehow, but I have not looked in the details.

**Comment by muhammadn** [comment]
@ahx We can handle it in the request, we don't need `x-tokeninfo-func` in this because this is specific to connexion. If we can add x-tokeninfo-func into the library for custom method it would be great but not compulsory.

https://spec.openapis.org/oas/v3.1.0

**Comment by muhammadn** [comment]
I've added a PR for `openapi-parser` for the initial work if you need to look at it. It looks a lot easier to read the code because it's arranged nicely to better understand the codebase.

https://github.com/ota42y/openapi_parser/pull/143/commits/5ffec5c6279e5175c61db8655a18dd4e57058b67

I took these below as my reference: 

https://swagger.io/docs/specification/authentication/bearer-authentication/
https://github.com/spec-first/connexion/blob/main/examples/jwt/spec/openapi.yaml

**Comment by ahx** [comment]
This will be addressed by https://github.com/ahx/openapi_first/issues/174
